azerothcore/azerothcore-wotlk
# feat(Core/SAI): implemented boss state handling in SMART_ACTION_SET_INST_DATA + added additional data checks

Well, this is minor port from trinitycore which allows SMART_ACTION_SET_INST_DATA action to set boss state (for instance boss data). Also additionally added improved data checks to ensure that valid data is loaded



## Changes Proposed:
-  SMART_ACTION_SET_INST_DATA handling for instance boss states (third param should be marked with 1
-  Additional logs to check SMART_ACTION_SET_INST_DATA action data
- This PR **DOES NOT** changes any of current SAI behaviour 

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Mainly been tested with unfinished Molten Core rewrite
- 


## How to Test the Changes:


1. You must select dungeon which supports boss states (AhnKahet or Obsidian Sanctum)
2. Perform this action with 3rd param set to 1. Param1 should contain boss id, Param2 should contain EncounterState value (see https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Instances/InstanceScript.h#L54)
3. Observe boss states with .instance getbossstate <bossId>

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
